lib/form_data.js

```diff
@@ -121,10 +121,28 @@ FormData.prototype._multiPartHeader = function(field, value, options) {
     header += '--' + boundary + FormData.LINE_BREAK +
       'Content-Disposition: form-data; name="' + field + '"';
 
+    // if the provided value is a Buffer treat the value as binary content
+    if (Buffer.isBuffer(value)) {
+      // let the user provide a filename for the buffer.
+      if(options.filename) {
+        header += '; filename="' + options.filename + '"';
+      }
+
+      header += FormData.LINE_BREAK;
+      // let the user set a custom content-type
+      if (options.contentType) {
+        header += 'Content-Type: ' + options.contentType;
+      } else {
+        header += 'Content-Type: application/octet-stream';
+      }
+
+      // if a buffer is being sent then set the encoding as binary.
+      header += FormData.LINE_BREAK + 'Content-Transfer-Encoding: binary';
+
     // fs- and request- streams have path property
     // or use custom filename and/or contentType
     // TODO: Use request's response mime-type
-    if (options.filename || value.path) {
+    } else if (options.filename || value.path) {
       header +=
         '; filename="' + path.basename(options.filename || value.path) + '"' + FormData.LINE_BREAK +
         'Content-Type: ' +  (options.contentType || mime.lookup(options.filename || value.path));
```

Comment:
Here should be change to `options.contentType || mime.lookup(options.filename || value.path) || 'application/octet-stream'`.

`mime.lookup` will return `false` when lookup falied, then data will be `'Content-Type: false'`